### PR TITLE
[modbus] Minor text updates

### PR DESCRIPTION
**Pull request by Rossko57**
A few small corrections and improvements.

I understand real names are required - I am Ross Kennedy

### CODE REVIEW

File: addons/binding/org.openhab.binding.modbus/README.md [hunk 1]
```diff
@@ -320,21 +317,23 @@ Main documentation on `autoupdate` in [Items section of openHAB docs](https://ww
...
-> * discrete input numbers start with a one and then span from 10001 to 19999
-> * input register numbers start with a three and then span from 30001 to 39999
-> * holding register numbers start with a four and then span from 40001 to 49999
+> * coils numbers start at 1, with a leading zero and then span from 00001 to 09999
+> * discrete input numbers start with a leading one and then span from 10001 to 19999
+> * input register numbers start with a leading three and then span from 30001 to 39999
+> * holding register numbers start with a leading four and then span from 40001 to 49999
 >
 > This translates into [entity] addresses between 0 and 9,998 in data frames.
+> Note that entity starts counting at 1, data frame address at 0.
```
Comment (ssalonen) [inline addons/binding/org.openhab.binding.modbus/README.md]: These are good changes I think. However, they are no longer direct quotes from Wikipedia. Should we remove the quoting (>) and change the way we refer to source

Comment (Rossko57) [reply addons/binding/org.openhab.binding.modbus/README.md]: I did not notice it was a quote.  In which case, just leave out this tiny change.

**Comment by Rossko57** [comment]
This long readme really needs some internal # hyperlinks, i.e. an clickable index of sections at the top.  So that sections can be referenced directly.
Seems like a good opportunity to do that also, but I do not have the skills with this editor.  Help!

File: addons/binding/org.openhab.binding.modbus/README.md [hunk 2]
```diff
@@ -524,7 +524,7 @@ Note that transformation is only one part of the overall process how polled data
 Consult [Read steps](#read-steps) and [Write steps](#write-steps) for more details.
 Specifically, note that you might not need transformations at all in some uses cases.
 
-Please also note that you should install relevant transformations, as necessary. For example, `openhab-transformation-javascript` feature provides the javascript (`JS`) transformation.
+Please also note that you should install relevant transformations in openHAB as necessary. For example, `openhab-transformation-javascript` feature provides the javascript (`JS`) transformation.
```
Comment (cweitkamp) [inline addons/binding/org.openhab.binding.modbus/README.md]:
```suggestion
Please also note that you should install relevant transformations in openHAB as necessary.
For example, `openhab-transformation-javascript` feature provides the JavaScript (`JS`) transformation.
```

**Comment by davidgraeff** [comment]
This is a general improvement, so I'll merge now.
I'll make a second PR for the @cweitkamp suggested change.

File: addons/binding/org.openhab.binding.modbus/README.md (hunk 2 above)
Comment (cweitkamp) [reply addons/binding/org.openhab.binding.modbus/README.md]: @davidgraeff Please don't get me wrong. The main reason for this suggestion was the incorrect spelling of JavaScript. Next time I will point it out more clearly. :wink:

**Pull request merged**